src/main/java/io/vertx/core/net/impl/TCPServerBase.java

```diff
@@ -162,6 +162,26 @@ public Future<Boolean> updateSSLOptions(ServerSSLOptions options, boolean force)
     }
   }
 
+  public void updateTrafficShapingOptions(TrafficShapingOptions options) {
+    if (options == null) {
+      throw new IllegalArgumentException("Invalid null value passed for traffic shaping options update");
+    }
+    if (trafficShapingHandler == null) {
+      throw new IllegalArgumentException("Unable to update traffic shaping options because the server was not configured " +
+                                         "to use traffic shaping during startup");
+    }
+    TCPServerBase server = actualServer;
+    if (server != null && server != this) {
+      server.updateTrafficShapingOptions(options);
+    } else {
+      if (options.getCheckIntervalForStatsTimeUnit() != null) {
+        trafficShapingHandler.configure(options.getOutboundGlobalBandwidth(), options.getInboundGlobalBandwidth(), options.getCheckIntervalForStats());
```

Comment:
This seems incorrect, we should probably compute the millis value for the check interval for stats property:

```
long checkIntervalForStatsInMillis = options.getCheckIntervalForStatsTimeUnit().toMillis(options.getCheckIntervalForStats());
```

I think we are missing updating the `maxDelayToWaitInMillis` value as well as the `maxGlobalWriteSize` 

```
    if (options.getPeakOutboundGlobalBandwidth() != 0) {
      trafficShapingHandler.setMaxGlobalWriteSize(options.getPeakOutboundGlobalBandwidth());
    }
```

Reply:
Thanks, addressed above